api/test/api-common.test.js

```diff
@@ -5554,6 +5608,151 @@ describe('API', () => {
+          expect(data.count).toBe(ordersExcludingAnyApp.length);
+          expect(data.orders).toBeDefined();
+          expect(Array.isArray(data.orders)).toBe(true);
+          expect(data.orders.length).toBe(ordersExcludingAnyApp.length);
+          data.orders.forEach((e) => {
+            expect(
+              e.order.apprestrict === allowedApp ||
+                e.order.apprestrict !== utils.NULL_ADDRESS,
+            ).toBe(true);
+          });
+        });
+
+        test('GET /datasetorders (workerpool filter & isWorkerpoolStrict): should exclude orders with "any" filtered workerpool)', async () => {
+          const { data, status } = await request
+            .get(
+              buildQuery('/datasetorders', {
+                chainId, // *
+                dataset: datasetAddress, // *
+                workerpool: allowedWorkerpool,
+                isWorkerpoolStrict: true,
+              }),
+            )
+            .then(parseResult);
+
+          const ordersExcludingAnyWorkerpool = workerpoolAllowedOrders.filter(
+            (order) => order.order.workerpoolrestrict !== utils.NULL_ADDRESS,
+          );
+          expect(status).toBe(OK_STATUS);
+          expect(data.ok).toBe(true);
+          expect(data.count).toBe(ordersExcludingAnyWorkerpool.length);
+          expect(data.orders).toBeDefined();
+          expect(Array.isArray(data.orders)).toBe(true);
+          expect(data.orders.length).toBe(ordersExcludingAnyWorkerpool.length);
+          data.orders.forEach((e) => {
+            expect(
+              e.order.workerpoolrestrict === allowedWorkerpool ||
+                e.order.workerpoolrestrict === utils.NULL_ADDRESS,
+            ).toBe(true);
+          });
+        });
+
+        test('GET /datasetorders (requester filter & isRequesterStrict): should exclude orders with "any" filtered requester)', async () => {
+          const { data, status } = await request
+            .get(
+              buildQuery('/datasetorders', {
+                chainId, // *
+                dataset: datasetAddress, // *
+                requester: allowedRequester,
+                isRequesterStrict: true,
+              }),
+            )
+            .then(parseResult);
+
+          const ordersExcludingAnyRequester = requesterAllowedOrders.filter(
+            (order) => order.order.requesterrestrict !== utils.NULL_ADDRESS,
+          );
+          expect(status).toBe(OK_STATUS);
+          expect(data.ok).toBe(true);
+          expect(data.count).toBe(ordersExcludingAnyRequester.length);
+          expect(data.orders).toBeDefined();
+          expect(Array.isArray(data.orders)).toBe(true);
+          expect(data.orders.length).toBe(ordersExcludingAnyRequester.length);
+          data.orders.forEach((e) => {
+            expect(
+              e.order.workerpoolrestrict === allowedWorkerpool ||
+                e.order.workerpoolrestrict === utils.NULL_ADDRESS,
+            ).toBe(true);
```

Comment:
same + bad copy/paste I guess workerpool => requester

```suggestion
            expect(e.order.requesterrestrict).toBe(allowedRequester);
```

Reply:
commit : 6fab488418a00e8ef7004ae32cd22009218e443c